corehq/middleware.py

```diff
@@ -123,8 +124,19 @@ def _session_expired(timeout, activity, time):
 
     @staticmethod
     def _user_requires_secure_session(couch_user):
-        return couch_user and any(Domain.is_secure_session_required(domain)
-                                  for domain in couch_user.get_domains())
+        if not couch_user:
+            return False
+
+        domains = couch_user.get_domains()
+        if any(Domain.is_secure_session_required(domain) for domain in domains):
+            return True
+
+        from corehq.apps.linked_domain.dbaccessors import get_linked_domains
+        for domain in domains:
+            if toggles.ENTERPRISE_LINKED_DOMAINS.enabled(domain):
```

Comment:
This means `secure_session` set on the enterprise applies to all linked domains?

Reply:
My thought was that the intention of this code is "return true if any domains the user has access to use secure sessions" and that enterprise users do have access to all linked domains. And that it's better to be more aggressive, rather than less aggressive, when it comes to secure sessions.

Reply:
I think [this comment](https://github.com/dimagi/commcare-hq/pull/10901#issuecomment-200444006) applies here.